### DISCUSSION TITLE
Fargate service's desired count reset during update?

### DISCUSSION THREAD

**Discussion by silvervest** [aws/aws-cdk#21615] [Q&A]
We recently had a deployment go wrong when a service that was updating as part of a deployment had the desired count reset to the minimum, due to the fact that the Service resource itself was replaced instead of updated. Previous updates to the service maintained the current desired count for the current load on the system, so we didn't expect this update to do the same and when it did our minimum (for overnight quiet period scale downs) wasn't enough to handle and we had a short outage.

The CDK docs note this, so we've used the default and not set manually.
```Default: - When creating the service, default is 1; when updating the service, default uses the current task number.`

This might be an obvious thing, but just checking if this could be a bug or just an implementation issue on our side? Anyone have an idea?

**Comment by dylanjgscott** [comment]
I'm also interested in this one :)
